### PR TITLE
Fixed shared_ptr comparisons with nullptr_t when spaceship is unavailable.

### PR DESCRIPTION
**Pull request by Bekenn**
@ldionne

This was causing compilation errors when attempting to compare a `shared_ptr<T[]>` with `nullptr`, as `get()` returns `T*` rather than `T (*)[]`.  `unique_ptr` did not have this issue, but I've added tests to make sure.

### CODE REVIEW

File: libcxx/test/std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.cmp/cmp_nullptr.pass.cpp
Comment (mordante) [inline libcxx/test/std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.cmp/cmp_nullptr.pass.cpp:90]: nowadays we typically use this style.
```suggestion
#if TEST_STD_VER >= 17
```
Please apply to similar cases.

**Review by mordante — CHANGES_REQUESTED** [review]
Thanks for your patch!

A few minor comments.

File: libcxx/test/std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.cmp/cmp_nullptr.pass.cpp
Comment (Bekenn) [reply libcxx/test/std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.cmp/cmp_nullptr.pass.cpp]: Updated.

File: libcxx/test/std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.cmp/cmp_nullptr.pass.cpp
Comment (Bekenn) [reply libcxx/test/std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.cmp/cmp_nullptr.pass.cpp:121]: Updated.

**Review by mordante — CHANGES_REQUESTED** [review]
The code LGTM. However the CI fails, I expect a rebase will fix the CI.
After rebasing I'll approve the CI run that need approval.

**Review by mordante — APPROVED** [review]
Thanks! LGTM! Do you need help landing this patch?

**Comment by Bekenn** [comment]
I don't currently have write access, though I have contributed elsewhere in llvm-project before, and have some other stuff I'm working on.

**Pull request merged**
